parsers/MD.py

```diff
@@ -263,16 +267,38 @@ def fetch_exchange(
         )
 
     archive_data = _get_archive_data(
-        target_datetime, session=session, num_backlog_days=1
+        target_datetime,
+        session=session,
+        num_backlog_days=num_backlog_days,
     )
 
-    exchange_list = ExchangeList(logger=logger)
+    use_actual = event_type == EventSourceType.measured
 
+    exchange_list = ExchangeList(logger=logger)
     for entry in archive_data:
         if sorted_zone_keys == ZoneKey("MD->UA"):
-            netflow = -1 * entry.exchange_UA_to_MD
+            netflow = (
+                entry.exchange_UA_to_MD
+                if use_actual
+                else entry.planned_exchange_UA_to_MD
+            )
+            if netflow is None:
+                continue
+
+            if netflow != 0:
+                netflow *= -1
+
         elif sorted_zone_keys == ZoneKey("MD->RO"):
-            netflow = -1 * entry.exchange_RO_to_MD
+            netflow = (
+                entry.exchange_RO_to_MD
+                if use_actual
+                else entry.planned_exchange_RO_to_MD
+            )
+            if netflow is None:
+                continue
+
+            if netflow != 0:
+                netflow *= -1
```

Comment:
There is a lot of duplicate code here, first of all we should instantiate the netflow variable outside the if statements. Then we can move out the 
```python
if netflow != 0:
    netflow *= -1
```
and only call it once.

This check should also be changed into:
```python
if netflow != 0 and not None:
    netflow *= -1
```

then we can remove the:
```python
if netflow is None:
    continue
```
checks as the parser class handles None values internally.

Reply:
Thanks for the suggestion! Something like this? 

```python
for entry in archive_data:

    if sorted_zone_keys == ZoneKey("MD->UA"):
        netflow = (
            entry.exchange_UA_to_MD
            if use_actual
            else entry.planned_exchange_UA_to_MD
        )

    elif sorted_zone_keys == ZoneKey("MD->RO"):
        netflow = (
            entry.exchange_RO_to_MD
            if use_actual
            else entry.planned_exchange_RO_to_MD
        )
    else:
        raise NotImplementedError(f"{sorted_zone_keys} pair is not implemented")

    if netflow != 0 and netflow is not None:
        netflow *= -1

    exchange_list.append(
        zoneKey=sorted_zone_keys,
        datetime=entry.datetime,
        netFlow=netflow,
        source=SOURCE,
        sourceType=event_type,
    )
```

Reply:
Exactly but you need to add `netflow = None` before the first if statement otherwise you can run into type errors with it not being defined.

Reply:
Perfect, added! (I thought the `raise` in the final catch-all `else:` block would have forced `netflow` to always be defined)

Reply:
It's quite possible it would, I didn't test it locally but I also know that some type checking systems have a hard time narrowing down types in some scenarios so better to just define it.

(I've been having related issues with that in TypeScript lately)